libvmaf/src/libvmaf.c

```diff
@@ -302,7 +306,11 @@ int vmaf_use_features_from_model(VmafContext *vmaf, VmafModel *model)
 
     for (unsigned i = 0; i < model->n_features; i++) {
         VmafFeatureExtractor *fex =
-            vmaf_get_feature_extractor_by_feature_name(model->feature[i].name, 1);
+#ifdef HAVE_CUDA
+            vmaf_get_feature_extractor_by_feature_name(model->feature[i].name, vmaf->cuda.state.ctx);
+#else
+            vmaf_get_feature_extractor_by_feature_name(model->feature[i].name, 0);
+#endif
```

Comment:
This does prevent a lazy cuda initialization. Cuda has to be initialized before selecting feature extractors.
@kylophone do you have an idea how to do this more elegant ? I would propose to add something like the `cpumask` to vmaf configuration struct which we could check here and could even throw as error when `vmaf_read_picture` is called with this config flag set, but `vmaf_cuda_init` not being called up to this point.

Reply:
> This does prevent a lazy cuda initialization. Cuda has to be initialized before selecting feature extractors.
> @kylophone do you have an idea how to do this more elegant ? I would propose to add something like the `cpumask` to vmaf configuration struct which we could check here and could even throw as error when `vmaf_read_picture` is called with this config flag set, but `vmaf_cuda_init` not being called up to this point.

Yes, I think something like that makes sense. Instead of matching on feature extractor name exactly, we should be matching on some sort of combination of hardware capabilities and gpumask.

Reply:
Roman proposed to make `vmaf_init` and `vmaf_cuda_init` mutually exclusive and set an internal state depending on this. Would you be ok with this too ?